src/adc.rs

```diff
@@ -14,27 +17,95 @@ use crate::{
         adc::{Channel as EmbeddedHalChannel, OneShot},
         blocking::delay::DelayUs,
     },
-    pac,
+    pac::{self, ADC1},
     rcc::{Enable, Reset, AHB2, CCIPR},
     signature::{VrefCal, VtempCalHigh, VtempCalLow, VDDA_CALIB_MV},
 };
 
-use pac::{ADC1, ADC_COMMON};
-use stable_deref_trait::StableDeref;
+/// Internal voltage reference channel, used for calibration.
+pub struct Vref {
+    _0: (),
```

Comment:
`_0` not needed in current Rust versions, use an empty type i.e. `pub struct Vref;`
There are multiple occurrences of this.

Reply:
I added this so it can only be constructed in this crate and not as a public API. Specifically, you can only get it with `enable_vref` if it isn't enabled yet. Same goes for `enable_vbat` and `enable_temperature`.